services/alertmanager/registry.go

```diff
@@ -115,6 +109,8 @@ func (r *Registry) collect() ammodels.PostableAlerts {
 	now := now()
 	for id, t := range r.times {
 		if t.Before(now) {
+			alert := r.alerts[id]
+			alert.EndsAt = strfmt.DateTime(now.Add(r.alertTTL))
```

Comment:
Seems like it is the right place for `EndsAt`. We want alert to live that time after we publish it.

Reply:
i still don't see how this will make the test pass, it will still fail because one of the fields in `time.Time` will differ, I think the problem is with how we use `now`, maybe we should focus on that and somehow try to provide in the same `now` value in the registry code and the test?

Reply:
I don't fully understand the problem. I tests we control value returned by `now()` so we control time used by tests and registry. Take a look at the last commit.